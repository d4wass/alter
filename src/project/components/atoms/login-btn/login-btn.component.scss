.login-btn {
  cursor: pointer;
  background-color: transparent;
  border: none;
}
