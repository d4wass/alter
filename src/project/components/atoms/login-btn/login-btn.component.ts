import { Component } from '@angular/core';
import { ModalLoginService } from '../../../services/modal-login/modal-login-service.service';

@Component({
  selector: 'app-login-btn',
  template: `
    <button class="login-btn" (click)="handleClick($event)">
      Log in
      <img src="assets/usericon.svg" alt="" />
    </button>
  `,
  styleUrls: ['./login-btn.component.scss']
})
export class LoginBtnComponent {
  constructor(private modalLoginService: ModalLoginService) {}

  handleClick(event: Event) {
    let target = (event.target as HTMLInputElement).className;
    this.modalLoginService.setModalVisibility(target === 'login-btn');
  }
}
