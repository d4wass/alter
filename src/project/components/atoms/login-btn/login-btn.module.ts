import { NgModule } from '@angular/core';
import { LoginBtnComponent } from './login-btn.component';
import { RouterModule } from '@angular/router';
import { CommonModule } from '@angular/common';

@NgModule({
  declarations: [LoginBtnComponent],
  imports: [RouterModule, CommonModule],
  exports: [LoginBtnComponent]
})
export class LoginBtnModule {}
