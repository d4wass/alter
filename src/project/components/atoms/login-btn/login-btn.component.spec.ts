import { DebugElement } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { MockComponent, MockProvider } from 'ng-mocks';
import { BehaviorSubject } from 'rxjs';
import { ModalLoginService } from '../../../../services/modal-login/modal-login-service.service';
import { LoginBtnComponent } from './login-btn.component';

describe('LoginBtnComponent', () => {
  let component: LoginBtnComponent;
  let fixture: ComponentFixture<LoginBtnComponent>;
  let btn: DebugElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [MockComponent(LoginBtnComponent)],
      providers: [
        MockProvider(ModalLoginService, {
          isVisible$: new BehaviorSubject(false),
          setModalVisibility: (isVisible) => isVisible$.next(isVisible)
        })
      ]
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(LoginBtnComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();

    btn = fixture.debugElement.query(By.css('.login-btn'));
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match to snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  describe('', () => {
    it('should call handleClick() on click', () => {
      jest.spyOn(component, 'handleClick');
      btn.triggerEventHandler('click', { target: { className: 'login-btn' } });

      expect(component.handleClick).toHaveBeenCalled();
    });

    it('should on handleClick call setModalvisibility in ModalLoginService', () => {
      component.handleClick = jest.fn();

      // expect(fixture.)
    });
  });
});
