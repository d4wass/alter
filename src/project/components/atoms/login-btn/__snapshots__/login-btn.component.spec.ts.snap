// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginBtnComponent should match to snapshot 1`] = `
<app-login-btn
  __ngContext__={[Function LRootView]}
  modalLoginService={[Function ModalLoginService]}
>
  <button
    class="login-btn"
  >
     Log in 
    <img
      alt=""
      src="assets/usericon.svg"
    />
  </button>
</app-login-btn>
`;
